towerlib/entities/inventory_script.py

```diff
@@ -0,0 +1,152 @@
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Main code for inventory_script.
+
+.. _Google Python Style Guide:
+   http://google.github.io/styleguide/pyguide.html
+
+"""
+
+import logging
+
+from towerlib.towerlibexceptions import InvalidValue
+from .core import (Entity,
+                   validate_max_length)
+
+__author__ = '''Yorick Hoorneman <yhoorneman@schubergphilis.com>'''
+__docformat__ = '''google'''
+__date__ = '''2018-01-03'''
+__copyright__ = '''Copyright 2018, Yorick Hoorneman'''
+__credits__ = ["Yorick Hoorneman"]
+__license__ = '''MIT'''
+__maintainer__ = '''Yorick Hoorneman'''
+__email__ = '''<yhoorneman@schubergphilis.com>'''
+__status__ = '''Development'''  # "Prototype", "Development", "Production".
+
+# This is the main prefix used for logging
+LOGGER_BASENAME = '''inventory_script'''
+LOGGER = logging.getLogger(LOGGER_BASENAME)
+LOGGER.addHandler(logging.NullHandler())
+
+
+class InventoryScript(Entity):
+    """Models the inventory script entity of ansible tower."""
+
+    def __init__(self, tower_instance, data):
+        Entity.__init__(self, tower_instance, data)
+
+    @property
+    def name(self):
+        """The name of the inventory script.
+
+        Returns:
+            string: The name of the inventory script.
+
+        """
+        return self._data.get('name')
+
+    @name.setter
+    def name(self, value):
+        """Update the name of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        max_characters = 512
+        conditions = [validate_max_length(value, max_characters)]
+        if all(conditions):
+            self._update_values('name', value)
+        else:
+            raise InvalidValue(('{value} is invalid. Condition max_characters must be '
+                                'less or equal to {max_characters}').format(value=value,
+                                                                            max_characters=max_characters))
+
+    @property
+    def description(self):
+        """The description of the inventory script.
+
+        Returns:
+            string: The description of the inventory script.
+
+        """
+        return self._data.get('description')
+
+    @description.setter
+    def description(self, value):
+        """Update the first name of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        self._update_values('description', value)
+
+    @property
+    def script(self):
+        """The script of the inventory script.
+
+        Returns:
+            string: The script of the inventory script.
+
+        """
+        return self._data.get('script')
+
+    @script.setter
+    def script(self, value):
+        """Update the script of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        self._update_values('script', value)
+
+    def create_inventory_script(self, name, description, script, organization):
```

Comment:
So an inventory script can create an inventory script? Is recursion intentional?

Reply:
The inventory_script template consists of many attributes, like name, description, and script. This is the setter for the script part. I believe this is correct.